packages/flutter/lib/src/material/chip.dart

```diff
@@ -2357,7 +2357,8 @@ class _RenderChip extends RenderBox {
         final Paint darkenPaint = Paint()
           ..color = selectionScrimTween.evaluate(checkmarkAnimation)
           ..blendMode = BlendMode.srcATop;
-        context.canvas.drawRect(avatarRect, darkenPaint);
+        final double shortestSide = math.min(avatarRect.width, avatarRect.height);
+        context.canvas.drawRRect(RRect.fromRectAndRadius(avatarRect, Radius.circular(shortestSide / 2)), darkenPaint);
```

Comment:
I assume this code will work as expected if the avatar's width and height are the same and it paints within a circle.

What if avatarRect is a rectangle and the avatar paints in a circle that's centered within its rect. Will this RRect line up with the avatar?